Sources/Commands/PackageTools/SwiftPackageTool.swift

```diff
@@ -126,14 +126,38 @@ extension SwiftPackageTool {
                 throw ValidationError("\(matchingPlugins.count) plugins found for '\(command)'")
             }
 
+            // handle plugin execution arguments that got passed after the plugin name
+            let pluginArguments = try PluginArguments.parse(Array(self.remaining.drop(while: { $0 == command })))
+            // merge the relevant plugin execution options
+            var pluginOptions = self._pluginOptions
+            pluginOptions.allowWritingToPackageDirectory = pluginOptions.allowWritingToPackageDirectory || pluginArguments.pluginOptions.allowWritingToPackageDirectory
+            pluginOptions.additionalAllowedWritableDirectories.append(contentsOf: pluginArguments.pluginOptions.additionalAllowedWritableDirectories)
+            if pluginArguments.pluginOptions.allowNetworkConnections != .none {
+                pluginOptions.allowNetworkConnections = pluginArguments.pluginOptions.allowNetworkConnections
+            }
+            swiftTool.shouldDisableSandbox = swiftTool.shouldDisableSandbox || pluginArguments.globalOptions.security.shouldDisableSandbox
+
             // At this point we know we found exactly one command plugin, so we run it.
             try PluginCommand.run(
                 plugin: matchingPlugins[0],
                 package: packageGraph.rootPackages[0],
                 packageGraph: packageGraph,
                 options: pluginOptions,
-                arguments: Array( remaining.dropFirst()),
-                swiftTool: swiftTool)
+                arguments: pluginArguments.remaining,
```

Comment:
I think you want to keep passing the same `Array(remaining.dropFirst())` arguments to the plugin, since the new approach is going to strip out anything that matches a `swift package` flag or option. Like `--verbose` could get eaten up during the re-parsing even if it has been making its way to the plugin.